test/request.js

```diff
@@ -92,5 +107,47 @@ describe('request', () => {
         responseStub.emit('end');
     });
 
+    ['gzip', 'deflate'].forEach((encoding) => {
+        it(`should inflate response body with ${encoding} encoding`, (done) => {
```

Comment:
Extracted `BufferStream` but left the `headers` as is because it felt weird to me to put it in `BufferStream`.